### PR TITLE
Plotting functions

### PR DESCRIPTION
**Pull request by ben-heil**
This PR has two main portions. The first is the creation of `plotting_util.py`, a file full of functions for plotting learning curves from results dictionaries returned by `classifier.py`. The second is the analysis `Evaluate_IRM_On_Sepsis.ipynb` which test drives the new plotting functions and compares the performance of training with invariant risk minimization to normal training.

Changes:
- Creation of `plotting_util.py`
- Creation of `0.Evaluate_IRM_On_Sepsis.ipynb `
- Memory leak fix in `classifier.py`

### CODE REVIEW

File: whistl/plot_util.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+''' This file contains functions for use in plotting loss curves from the results returned by
+classifier.py. I decided to create a different plotting function for each metric both to make
+plotting individual metrics super simple (just pass in results), and to allow consistant plotting
+style across analyses.
+'''
```
Comment (arielah) [inline whistl/plot_util.py:5]: Good note. I was just about to ask why all the functions were separate, so it's good to have a justification in there.

File: analyses/0.Evaluate_IRM_On_Sepsis.ipynb [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
...v0tttaLJuR63O0TTIVmHnr06GFQOlZWVoIzttasWQN/f3+tcM1ZSIZspHjy5Emdx7n7cCkUCnYhREDbbK8pLpo3by6YppDPjiZSHZRNMaOqJPjllCRMtTdWeYAOYxkOFTsUi6HLspORkYEBAwYIOr9GRUVhypQpmDp1quC5mg63ZE8jMlSUlJTEW59m/PjxGDFiBOLj4zF48GC888477LHly5fjp59+4qX31VdfYdq0aTpXRJ02bRrCw8NRt25d9OrVC126dGGvZWrEvuimTJkiKx2xlYCnTp2KNm3aYPTo0ZLiA0UC7Ny5czqvd+/ePVaA3rlzB3fv3sV7773HixMUFITVq1cD0O74NIegpk6dinv37rHHFy1apDU1H+ALGbHOlFiFuFt3CCEkdky9pYgcyoI4MPYeqM8ORQrUDkYxO1KmfJ88eRI7duzA559/jtq1awMAHjx4ACsrK95u0UKOzf/88w/7t0KhYKdWb926FUuWLGEXjiMkJSUhKSkJFSpU0Fqo7tSpU4L5E1rkj8uGDRvQtm1bpKWlscMuwH9DPS4uLnj58qXONKRCOtyOHTvi2LFjbLiuxu+LL75AaGgoAgMDJcWvUKGC1m7nFSpUQGpqKi+sS5cuOHLkCFxdXQW3YuDi6OiI6OhoKBQK1hFVaAFHMpRBGvYDBw7A1dVVK78qlQqOjo7sbysrK0Eh+OrVK/ZvsQ5xw4YNiI+P5/kUkbhbt27F0qVLcfz4ccFzly5dyquDZZ1z586ZdJYiXX9IOtQKZji0llEshlwHZX9/fzRp0kRvPE1rkabjrJhFwpD1ZsTIy8tD3759RY9HRESY7Fqk09e0vOiyPL333nuoX78+z+dGc9sDLikpKVo7hQvtHN6/f38A0p9ts2bNeOuDkGc1duxY1iFZcx+ggIAA1KlTx+AvWSlip2rVqujYsaPg8Y4dO7LOz0Idc+XKlREcHGxQ3kojvr6+CAgIMFl61LIjHWrZMRwqdiiymThxokHOmOZyUNacbaMpdsQ6dVNtC0H8g8SmVQvtVm4MpJHjWjWAogX5yPCZvb097zixWJBOG9Dt4JiXl6c1PCN0fyRdQ2faEf+ob775BjNnzgTwn6DQLDNDG/eGDRuyf+t7Dvq+nE3xHKnPDp/yIFJMhdBq07Q+SYOKHYpsNmzYgPv378s+z5xTzwmbN2/mOQXPnDlT9
```
Comment (arielah) [inline analyses/0.Evaluate_IRM_On_Sepsis.ipynb:408]: I think there's a word missing in the sentence below this, "The learning curve when training normally, but the tuning set accuracy doesn't reach the same level as in IRM."

Comment (arielah) [inline analyses/0.Evaluate_IRM_On_Sepsis.ipynb:307]: I would add another comment here, if only to break up the plots from irm and erm. It's hard to tell if your comment at the top of 'Results' applies to some or all of the plots, and me being an idiot who doesn't know the context, it makes it hard to follow.

File: whistl/plot_util.py (hunk 1 above)
Comment (arielah) [reply whistl/plot_util.py:5]: It still seems like you could have these all be one or a couple of functions by just passing in an additional couple of parameters, but that would probably take a lot of reworking for a modest reduction of code duplication so I understand why not if you don't.

**Review by arielah — APPROVED** [review]
Looks good! I made a couple of suggestions to increase clarity, but it's very nicely written code.

File: whistl/plot_util.py (hunk 1 above)
Comment (ben-heil) [reply whistl/plot_util.py:5]: I agree. I imagine I'll call these functions frequently, so increase in simplicity in the calling code is worth the increase in complexity here

File: analyses/0.Evaluate_IRM_On_Sepsis.ipynb (hunk 2 above)
Comment (ben-heil) [reply analyses/0.Evaluate_IRM_On_Sepsis.ipynb:408]: Good catch!

Comment (ben-heil) [reply analyses/0.Evaluate_IRM_On_Sepsis.ipynb:307]: Good call, the results heading was too big and too ambiguous, so I made it smaller and changed it to 'IRM results' or 'ERM results'
